cupyx/scipy/interpolate/_interpolate.py

```diff
@@ -294,6 +295,132 @@
         [f'fix_continuity<{type_name}>' for type_name in TYPES] +
         [f'integrate<{type_name}>' for type_name in TYPES]))
 
+BPOLY_KERNEL = BASE_HEADERS + r"""
+#include <cupy/complex.cuh>
+#include <cupy/math_constants.h>
+
+template<typename T>
+__device__ T eval_bpoly1(
+        const double s, const T* coef, const long long ci, const long long cj,
+        const long long* c_dims, const long long* c_strides) {
+
+    const long long k = c_dims[0] - 1;
+    const double s1 = 1 - s;
+    T res;
+
+    const long long i0 = 0 * c_strides[0] + ci * c_strides[1] + cj;
+    const long long i1 = 1 * c_strides[0] + ci * c_strides[1] + cj;
+    const long long i2 = 2 * c_strides[0] + ci * c_strides[1] + cj;
+    const long long i3 = 3 * c_strides[0] + ci * c_strides[1] + cj;
+
+    if(k == 0) {
+        res = coef[i0];
+    } else if(k == 1) {
+        res = coef[i0] * s1 + coef[i1] * s;
+    } else if(k == 2) {
+        res = coef[i0] * s1 * s1 + coef[i1] * 2.0 * s1 * s + coef[i2] * s * s;
+    } else if(k == 3) {
+        res = (coef[i0] * s1 * s1 * s1 + coef[i1] * 3.0 * s1 * s1 * s +
+               coef[i2] * 3.0 * s1 * s * s + coef[i3] * s * s * s);
+    } else {
+        T comb = 1;
+        res = 0;
+        for(int j = 0; j < k + 1; j++) {
+            const long long idx = j * c_strides[0] + ci * c_strides[1] + cj;
+            res += comb * pow(s, j) * pow(s1, ((int) k) - j) * coef[idx];
```

Comment:
@takagi, do you know if `pow` is valid here? If I try to compile the kernel, then the compiler will complain about not finding an adequate symbol definition for `pow`:

```python
  File "/home/andfoy/Documentos/Quansight/cupy/cupy/cuda/compiler.py", line 322, in compile_using_nvrtc
    return _compile(source, options, cu_path,
  File "/home/andfoy/Documentos/Quansight/cupy/cupy/cuda/compiler.py", line 306, in _compile
    compiled_obj, mapping = prog.compile(options, log_stream)
  File "/home/andfoy/Documentos/Quansight/cupy/cupy/cuda/compiler.py", line 705, in compile
    raise CompileException(log, self.src, self.name, options,
cupy.cuda.compiler.CompileException: ptxas fatal   : Unresolved extern function '_Z4powidi'
```

If I try to use either `-dc` or `-rdc=true`, then the kernel will be marked as invalid:

```python
File "cupy/_core/core.pyx", line 2265, in cupy._core.core.compile_with_cache
    return cuda.compiler._compile_module_with_cache(
  File "/home/andfoy/Documentos/Quansight/cupy/cupy/cuda/compiler.py", line 493, in _compile_module_with_cache
    return _compile_with_cache_cuda(
  File "/home/andfoy/Documentos/Quansight/cupy/cupy/cuda/compiler.py", line 580, in _compile_with_cache_cuda
    cubin = ls.complete()
  File "cupy/cuda/function.pyx", line 306, in cupy.cuda.function.LinkState.complete
    cpdef bytes complete(self):
  File "cupy/cuda/function.pyx", line 307, in cupy.cuda.function.LinkState.complete
    cubin = driver.linkComplete(self.ptr)
  File "cupy_backends/cuda/api/driver.pyx", line 185, in cupy_backends.cuda.api.driver.linkComplete
    check_status(status)
  File "cupy_backends/cuda/api/driver.pyx", line 60, in cupy_backends.cuda.api.driver.check_status
    raise CUDADriverError(status)
cupy_backends.cuda.api.driver.CUDADriverError: CUDA_ERROR_INVALID_SOURCE: device kernel image is invalid
```

Reply:
`pow` only accepts `(double, double)` in CUDA:
https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__DOUBLE.html#group__CUDA__MATH__DOUBLE_1g6d36757715384dc18e0483aa1f04f6c7

c.f. https://github.com/cupy/cupy/blob/46f632218f7ccedffd7182eb37294229e25e3aee/cupy/_core/_routines_math.pyx#L1007-L1022

(thanks @emcastillo for the hint 😃)

Reply:
Thanks a lot for the pointers!